### PR TITLE
Add the case of server-threadpool-set for virt-admin

### PR DESCRIPTION
**Pull request by iccaszhulili**
Signed-off-by: Lily Zhu <lizhu@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import logging
+from avocado.core import exceptions
+from virttest import virt_admin
+from virttest import virsh
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: please remove this import to fix travis failure

Comment (iccaszhulili) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: @balamuruhans Thanks. This issue has been fixed.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                                               ignore_status=True, debug=True)
+            outdict = threadpool_info(server_name)
+            if result.exit_status:
+                raise exceptions.TestFail("This operation should success "
+                                          "but failed! output:\n%s ", result)
+            elif outdict["minWorkers"] != min_workers:
+                raise exceptions.TestFail("attributes set by server-threadpool-set "
+                                          "is not correct!")
+    finally:
+        config.restore()
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: variable `config` is not declared

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    libvirtd = utils_libvirtd.Libvirtd()
+    vp = virt_admin.VirtadminPersistent()
+
+    def threadpool_info(server):
+        """
+        check the attributes by server-threadpool-set.
+        1) get the output returned by server-threadpool-set;
+        2) split the output to get a dictionary of those attributes;
+        """
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: docstring for param and return value can be added

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        return out_dict
+    try:
+        if "min-workers" in options_ref:
+            if int(min_workers) < int(nworkers):
+                result = vp.srv_threadpool_set(server_name, min_workers=min_workers,
+                                               ignore_status=True, debug=True)
+            outdict = threadpool_info(server_name)
+            if result.exit_status:
+                raise exceptions.TestFail("This operation should success "
+                                          "but failed! output:\n%s ", result)
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]:
```
raise exceptions.TestFail("This operation should success "
                          "but failed! output:\n%s " % result)
```

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py (hunk 2 above)
Comment (waynesun09) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: @iccaszhulili pls fix this as it fail the testing:

2017-03-15 08:04:57,132 stacktrace       L0042 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py", line 44, in run
2017-03-15 08:04:57,132 stacktrace       L0042 ERROR|     config.restore()
2017-03-15 08:04:57,133 stacktrace       L0042 ERROR| NameError: global name 'config' is not defined

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py (hunk 3 above)
Comment (iccaszhulili) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: update. Thx very much.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py (hunk 4 above)
Comment (iccaszhulili) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: updated

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py (hunk 2 above)
Comment (iccaszhulili) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: @waynesun09 @balamuruhans updated. Thx very much.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        out_dict = dict([[item[0].strip(), item[1].strip()] for item in out_split])
+        return out_dict
+
+    try:
+        if "min-workers" in options_ref:
+            if int(min_workers) < int(nworkers):
+                result = vp.srv_threadpool_set(server_name, min_workers=min_workers,
+                                               ignore_status=True, debug=True)
+            outdict = threadpool_info(server_name)
+            if result.exit_status:
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: you can check exit_status of result inside `if int(min_workers) < int(nworkers):`, is there any reason to check it outside ?

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                raise exceptions.TestFail("This operation should success "
+                                          "but failed! output:\n%s" % result)
+            elif outdict["minWorkers"] != min_workers:
```
Comment (balamuruhans) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: variable `outdict` can be avoided as used only once,
```
if threadpool_info(server_name)["minWorkers"] != min_workers:
    raise ...
```

File: libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py (hunk 5 above)
Comment (iccaszhulili) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_threadpool_set.py]: @balamuruhans As the new test scenarios are added, you can see why I would check it outside.

**Comment by will-Do** [comment]
As no more comment, I'm gonna merge it

**Pull request merged**